<?xml version="1.0"?>
<launch>

    <!-- Arguments for Image Tracker -->
    <arg name="tracker_image_source_topic"            default="/realsense/color/image_raw" />
    <arg name="tracker_image_output_topic"            default="/tracked_frame" />
    <arg name="tracker_type"                          default="CSRT" />
    <arg name="tracker_bbox_topic"                    default="/tracked_obj_bbox" />

    <!-- Arguments for Object Point Cloud Extraction -->
    <arg name="extractor_input_point_cloud_topic"     default="/realsense/depth_registered/points" />
    <arg name="extractor_input_bbox_topic"            default="$(arg tracker_bbox_topic)" />
    <arg name="extractor_output_point_cloud_topic"    default="/tracked_object_point_cloud" />

    <!-- Arguments for template capture -->
    <arg name="capture_template"                      default="false" />
    <arg name="template_pointcloud_target_folder"     default="$(find arm_vs)/dataset/vitamin" />
    <arg name="template_pointcloud_filename_prefix"   default="ptcld_" />

    <!-- Image tracker -->
    <node name="object_image_tracking" pkg="arm_vs" type="arm_vs_obj_image_tracking" output="screen" >
        <param name="input_image_topic"     type="string" value="$(arg tracker_image_source_topic)" />
        <param name="output_image_topic"    type="string" value="$(arg tracker_image_output_topic)" />
        <param name="tracker_type"          type="string" value="$(arg tracker_type)" />
        <param name="output_bbox_topic"     type="string" value="$(arg tracker_bbox_topic)" />
    </node>

    <!-- Point cloud extractor -->
    <node name="object_pointcloud_extractor" pkg="arm_vs" type="arm_vs_obj_pointcloud_extraction" output="screen" >
        <param name="input_pointcloud_topic"    type="string" value="$(arg extractor_input_point_cloud_topic)" />
        <param name="input_bbox_topic"          type="string" value="$(arg extractor_input_bbox_topic)" />
        <param name="output_pointcloud_topic"   type="string" value="$(arg extractor_output_point_cloud_topic)" />
    </node>

    <node name="viewer" pkg="rqt_image_view" type="rqt_image_view" />

</launch>